### PR TITLE
Fix/1498/keep selection when rotating

### PR DESCRIPTION
**Pull request by JaiJoPa**
# Keep selection when rotation the map

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

Issue: #1498 

## Description

**Descriptive pull request text**, answering:
  - Selected Building will not be deselected when rotationg the map
  - When the left button is released, it is checked, if the mouse position has been moved. The clearfunction will only be called if the mouse has not been moved

## Screenshots or gifs
selctionBugfix

### CODE REVIEW

File: visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.spec.ts [hunk 1]
```diff
@@ -397,13 +397,21 @@ describe("codeMapMouseEventService", () => {
...
-				threeSceneService.getHighlightedBuilding = jest.fn()
+			it("should call clearselection, when the mouse has not moved while left button was pressed", () => {
+				threeSceneService["selected"] = codeMapBuilding
+				codeMapMouseEventService["hasMouseMoved"] = jest.fn().mockReturnValue(false)
 
 				codeMapMouseEventService.onDocumentMouseUp(event)
 
 				expect(threeSceneService.clearSelection).toHaveBeenCalled()
 			})
+			it("should not call clear selection, whe mouse has moved while left button was pressed", () => {
```
Comment (NearW) [inline visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.spec.ts:408]: - add empty line break between tests
- fix typo `whe mouse has moved` to `when mouse has moved`

File: visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts [hunk 2]
```diff
@@ -214,9 +214,9 @@ export class CodeMapMouseEventService
 	}
 
 	private onLeftClick() {
-		this.threeSceneService.clearSelection()
 		this.isGrabbing = false
 		if (!this.hasMouseMoved(this.mouseOnLastClick)) {
```
Comment (NearW) [inline visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts:218]: Do you mind adding a threshold to the `hasMouseMoved` function? Moving the mouse by 1px should maybe count as not moved at all. A threshold of 5px should be good enough for a better user experience.

**Review by NearW — APPROVED** [review]
LGTM otherwise

File: visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts (hunk 2 above)
Comment (JaiJoPa) [reply visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts]: Sure ! Good idea, added a threshold for the right and the left button of 3 px. If its to less, we can raise it.

**Review by BridgeAR — APPROVED** [review]

**Review by ce-bo — APPROVED** [review]

**Pull request merged**